### PR TITLE
[authproxy] add cusom header and group configuration documentation

### PR DESCRIPTION
**Pull request by seuf**
Update authproxy documentation with configurable headers and staticGroups.

See : https://github.com/dexidp/dex/pull/2371

### CODE REVIEW

**Comment by nabokihms** [comment]
@nate-double-u, could you please check this PR when you have time? This is a doc for the new feature.

**Comment by nate-double-u** [comment]
I think I can take a look at this on Monday 🙂

File: content/docs/connectors/authproxy.md [hunk 1]
```diff
@@ -16,16 +16,22 @@ The `authproxy` connector returns identities based on authentication which your
 front-end web server performs. Dex consumes the `X-Remote-User` header set by
 the proxy, which is then used as the user's email address.
 
+It also consume the `X-Remote-Group` header to use as the user's group.
```
Comment (nate-double-u) [inline content/docs/connectors/authproxy.md:19]:
```suggestion
It also consumes the `X-Remote-Group` header to use as the user's group.
```

File: content/docs/connectors/authproxy.md [hunk 2]
```diff
@@ -16,16 +16,22 @@ The `authproxy` connector returns identities based on authentication which your
...
+
+Header's names can be configured via the `userHeader` and `groupHeader` config.
+
+Additional static groups can also be defined in connector's configuration.
```
Comment (nate-double-u) [inline content/docs/connectors/authproxy.md:23]:
```suggestion
Additional static groups can also be defined in the connector's configuration.
```

File: content/docs/connectors/authproxy.md [hunk 3]
```diff
@@ -63,12 +69,23 @@ connectors:
...
+    - default
 ```
 
 The authproxy connector assumes that you configured your front-end web server
 such that it performs authentication for the `/dex/callback/myBasicAuth`
-location and provides the result in the X-Remote-User HTTP header. The following
-configuration will work for Apache 2.4.10+:
+location and provides the result in the HTTP headers.
+
+In this example, the configured headers are `X-Forwarded-User` for user's mail
```
Comment (nate-double-u) [inline content/docs/connectors/authproxy.md:83]:
```suggestion
In this example, the configured headers are `X-Forwarded-User` for the user's mail
```

File: content/docs/connectors/authproxy.md [hunk 4]
```diff
@@ -63,12 +69,23 @@ connectors:
...
+and `X-Forwarded-Group` for user's group.
```
Comment (nate-double-u) [inline content/docs/connectors/authproxy.md:84]:
```suggestion
and `X-Forwarded-Group` for the user's group.
```

File: content/docs/connectors/authproxy.md [hunk 5]
```diff
@@ -63,12 +69,23 @@ connectors:
...
+Dex authproxy connector will return a list of groups containing both
+configured `staticGroups` and returnded group header.
```
Comment (nate-double-u) [inline content/docs/connectors/authproxy.md:86]:
```suggestion
configured `staticGroups` and return the group header.
```

File: content/docs/connectors/authproxy.md [hunk 6]
```diff
@@ -16,16 +16,22 @@ The `authproxy` connector returns identities based on authentication which your
...
 
-The connector does not support refresh tokens or groups.
+The connector does not support refresh tokens.
 
 ## Configuration
 
 The `authproxy` connector is used by proxies to implement login strategies not
 supported by dex. For example, a proxy could handle a different OAuth2 strategy
-such as Slack. The connector takes no configuration other than a `name` and `id`:
+such as Slack :
```
Comment (nate-double-u) [inline content/docs/connectors/authproxy.md:34]:
```suggestion
such as Slack:
```

**Review by nate-double-u — APPROVED** [review]
I've done a copy edit pass.

It looks good to me once the inline comments are resolved.

**Comment by nate-double-u** [comment]
Thanks for the updates @seuf, unfortunately the GitHub UI doesn't make it obvious that the DCO is still required, even when accepting a suggestion.

To merge this, each commit needs to be signed. https://github.com/dexidp/website/pull/109/checks?check_run_id=5657854026

**Comment by seuf** [comment]
Hello, I've squashed the commits, so we have only one signed commit with all the changes

**Comment by nabokihms** [comment]
@seuf thank you for contributing!

**Pull request merged**